src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -402,6 +402,14 @@ public Intent getIntentForBrokerActivity(final AuthenticationRequest request) {
             if (intent != null) {
                 intent.putExtra(AuthenticationConstants.Broker.BROKER_REQUEST,
                         AuthenticationConstants.Broker.BROKER_REQUEST);
+                
+                // Only the new broker with PRT support can read the new PromptBehavior force_prompt. 
+                // If talking to the old broker, and PromptBehavior is set as force_prompt, reset it as 
+                // Always. 
+                if (!isBrokerWithPRTSupport(intent) && PromptBehavior.FORCE_PROMPT == request.getPrompt()) {
+                    Logger.v(TAG, "FORCE_PROMPT is set for broker auth via old version of broker app, reset to ALWAYS.");
+                    intent.putExtra(AuthenticationConstants.Broker.ACCOUNT_PROMPT, PromptBehavior.Always.name());
```

Comment:
@weijjia Why don't we always pass the prompt behavior to the broker.  In general, I think it would be best to just pass all of the request parameters and data to the broker and let it decide how to best make use of it.